cmd/auth-handler.go

```diff
@@ -153,7 +154,7 @@ func checkRequestAuthType(ctx context.Context, r *http.Request, action policy.Ac
 	var locationConstraint string
 	if action == policy.CreateBucketAction {
 		// To extract region from XML in request body, get copy of request body.
-		payload, err := ioutil.ReadAll(r.Body)
+		payload, err := ioutil.ReadAll(io.LimitReader(r.Body, maxLocationConstraintSize))
```

Comment:
What error does AWS S3 return for oversized request body for create bucket call?

Reply:
Fixed - PTAL

Reply:
@aead We are not sending 400 Bad Request if `r.Body` is > `maxLocationConstraintSize`,  If request body contains valid XML first `maxLocationConstraintSize` bytes and rest are junk, we continue servicing than returning an error.

We suppose to return error if `r.Body` > `maxLocationConstraintSize`